### PR TITLE
[GLOSSARY] Add Block filter

### PR DESCRIPTION
**Pull request by MaxHillebrand**
Adds BIP 158 to the glossary.

### CODE REVIEW

**Comment by MaxHillebrand** [comment]
I was experimenting with git here, I committed and pushed this #425 while #421 was commited but the PR was not merged.
Then I made the #425 branch ontop of the #421 branch, then made these changes to add block filter glossary, then pushed it.
It turns out that the two commits of #421 are now also in this branch, and I'm not sure if there are going to be any troubles when merging this branch...

File: docs/glossary/Glossary.md [hunk 1]
```diff
@@ -142,10 +142,20 @@ It's the quantity of equal value outputs of one CoinJoin transaction.
 Read more: [What is the anonimity set?](/FAQ/FAQ-UseWasabi.md#what-is-the-anonymity-set)
 :::
 
+:::details
+### Block filters
+
+A filter representing all addresses in one block.
```
Comment (yahiheb) [inline docs/glossary/Glossary.md]:
```suggestion
A filter representing a compact list of addresses in one block.
```

File: docs/glossary/Glossary.md [hunk 2]
```diff
@@ -142,10 +142,20 @@ It's the quantity of equal value outputs of one CoinJoin transaction.
...
+Wasabi checks locally if any block has transactions with addresses of this wallet.
```
Comment (yahiheb) [inline docs/glossary/Glossary.md]:
```suggestion
Wasabi checks locally if any block filter contains transactions with addresses of the wallet.
```

File: docs/glossary/Glossary.md [hunk 3]
```diff
@@ -142,10 +142,20 @@ It's the quantity of equal value outputs of one CoinJoin transaction.
...
+No public keys are send to a third party server, thus it is very private.
```
Comment (yahiheb) [inline docs/glossary/Glossary.md]:
```suggestion
No public keys are sent to any third party server, thus it is very private.
```

File: docs/glossary/Glossary.md [hunk 4]
```diff
@@ -142,10 +142,20 @@ It's the quantity of equal value outputs of one CoinJoin transaction.
...
+Read more: [BIP 158: Compact Block Filters for Light Clients](/using-wasabi/BIPs.md#bip-158-compact-block-filters-for-light-clients)
+:::
+
 :::details
 ### Bloom Filter
 
 A filter used primarily by SPV clients to request only block headers and merkle proofs of a given transaction from full nodes.
+This is very bad for privacy, as a third party learns about which addresses you are interested in.
```
Comment (yahiheb) [inline docs/glossary/Glossary.md]:
```suggestion
This is very bad for privacy, as third party servers learn about which addresses you are interested in.
```

**Comment by yahiheb** [comment]
I think this PR can't be merged.

**Comment by RiccardoMasutti** [comment]
@MaxHillebrand ?

**Pull request merged**